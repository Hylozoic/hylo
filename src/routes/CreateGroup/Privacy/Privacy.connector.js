import { connect } from 'react-redux'
import { push } from 'connected-react-router'
import { get } from 'lodash/fp'
import { addGroupPrivacy } from '../CreateGroup.store'

export function mapStateToProps (state, props) {
  return {
    groupPrivacy: get('privacy', state.CreateGroup)
  }
}

export function mapDispatchToProps (dispatch, props) {
  return {
    goToNextStep: () => dispatch(push('/create-group/review')),
    goToPreviousStep: () => dispatch(push('/create-group/domain')),
    addGroupPrivacy: (privacy) => dispatch(addGroupPrivacy(privacy)),
    goHome: () => dispatch(push('/'))
  }
}

export default connect(mapStateToProps, mapDispatchToProps)
