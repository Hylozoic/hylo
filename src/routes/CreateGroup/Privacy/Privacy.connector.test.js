import { mapDispatchToProps, mapStateToProps } from './Privacy.connector'

const dispatch = jest.fn(x => x)
const props = {}
const dispatchProps = mapDispatchToProps(dispatch, props)

describe('Privacy', () => {
  it('should call addGroupPrivacy from mapDispatchToProps', () => {
    const privacy = 'privacy'
    expect(dispatchProps.addGroupPrivacy(privacy)).toMatchSnapshot()
  })

  it('should call goToPreviousStep from mapDispatchToProps', () => {
    expect(dispatchProps.goToPreviousStep()).toMatchSnapshot()
  })

  it('should call goToNextStep from mapDispatchToProps', () => {
    expect(dispatchProps.goToNextStep()).toMatchSnapshot()
  })

  it('should call goHome from mapDispatchToProps', () => {
    expect(dispatchProps.goHome()).toMatchSnapshot()
  })

  it('should have groupPrivacy in mapStateToProps', () => {
    const privacy = 'privacy'

    const state = {
      CreateGroup: {
        privacy
      }
    }
    expect(mapStateToProps(state, props).groupPrivacy).toBe(privacy)
  })
})
