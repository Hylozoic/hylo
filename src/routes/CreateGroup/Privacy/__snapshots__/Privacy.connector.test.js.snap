// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Privacy should call addGroupPrivacy from mapDispatchToProps 1`] = `
Object {
  "payload": "privacy",
  "type": "CreateGroup/ADD_GROUP_PRIVACY",
}
`;

exports[`Privacy should call goHome from mapDispatchToProps 1`] = `
Object {
  "payload": Object {
    "args": Array [
      "/",
    ],
    "method": "push",
  },
  "type": "@@router/CALL_HISTORY_METHOD",
}
`;

exports[`Privacy should call goToNextStep from mapDispatchToProps 1`] = `
Object {
  "payload": Object {
    "args": Array [
      "/create-group/review",
    ],
    "method": "push",
  },
  "type": "@@router/CALL_HISTORY_METHOD",
}
`;

exports[`Privacy should call goToPreviousStep from mapDispatchToProps 1`] = `
Object {
  "payload": Object {
    "args": Array [
      "/create-group/domain",
    ],
    "method": "push",
  },
  "type": "@@router/CALL_HISTORY_METHOD",
}
`;
